### ISSUE TITLE
Glacier Example Getting Access Denied

### ISSUE THREAD

**Issue by chad-skidmore** [chilts/awssum#59]
I have been playing with your addition for Glacier since yesterday and am continuing to get a 403 access denied. I've tried with two different AWS accounts and verified that the user and keys I am using have access and are signed up for Glacier. I am able to create/delete vaults via the AWS console but unable to use the simple listvaults example. Not sure where to go on debugging other than possibly something in the signature process. Any thoughts?

Output is as follows (Account ID removed):

Region               : us-west-2
EndPoint             : glacier.us-west-2.amazonaws.com
AccessKeyId          : AKI...
SecretAccessKey      : OUD...
AwsAccountId         : XXX
listing vaults - expecting success
Error : { StatusCode: 403,
  Headers: 
   { 'x-amzn-requestid': 'twIBNMXzwPf18LgEefj2591vqonJmfGXz6f9pn96EjCAZJw',
     'content-type': 'application/json',
     'content-length': '103',
     date: 'Wed, 22 Aug 2012 22:53:26 GMT' },
  Body: 
   { message: 'Access denied for account ID: XXX',
     code: 'AccessDeniedException',
     type: 'Client' } }
Data : null

**Comment by chilts** [comment]
Hi Chad,

So sorry about that. I mis-read what was needed for CreateVault and obviously didn't check what I did!!! My bad. :)

Thanks for the heads up and I hope it's all okay now. Am glad you're trying Glacier ... and I hope you don't run into more problems, but don't hesitate to shout if you find anything else. :)

I'm afraid you're the (pioneer/guinea pig)*. :)

(*) pick one.

Cheers,
Andy

**Comment by chilts** [comment]
Just added an example for CreateVault:
- https://github.com/appsattic/node-awssum/blob/master/examples/amazon/glacier/create-vault.js

Andy :)

**Comment by chad-skidmore** [comment]
Thanks for the crazy fast response Andy. I think you misunderstood however. I'm seeing issues with each of the Glacier operations. Seems to be authentication issue as I consistently get back a 403 StatusCode. I originally thought that it was a key/secret issue or that my account was not setup for Glacier but I believe I've validated that as I can use the AWS console for Glacier without issue. I've also tried a different AWS account and tried new keys as well. Are you sure the signature and auth process is working correctly? I could easily be doing something wrong as well but I am running your examples as written with the addition of proper env variables for my account details.

**Comment by chilts** [comment]
Hi Chad,

Just to check, did you clone the master branch of the repo? I know that's a silly question, but I just need to make sure. As I say, I did actually fix something to do with ListVaults, so you'll need the latest commit.

When I run ListVaults, it is running fine here:

```
$ node amazon/glacier/list-vaults.js
Region               : us-east-1
EndPoint             : glacier.us-east-1.amazonaws.com
AccessKeyId          : XXX
SecretAccessKey      : XXX
AwsAccountId         : XXX
listing vaults - expecting success
Error : null
Data : { StatusCode: 200,
  Headers: 
   { 'x-amzn-requestid': 'JzvVrG268S9faeAz0xGResKMEglM0Ov2FgFXc77-tSpYXms',
     'content-type': 'application/json',
     'content-length': '230',
     date: 'Thu, 23 Aug 2012 00:28:19 GMT' },
  Body: 
   { Marker: null,
     VaultList: 
      [ { CreationDate: '2012-08-22T23:08:16.884Z',
          LastInventoryDate: null,
          NumberOfArchives: 0,
          SizeInBytes: 0,
          VaultARN: 'arn:aws:glacier:us-east-1:XXX:vaults/test-vault',
          VaultName: 'test-vault' } ] } }
```

If, as you say, you can run those things from other programs, then I'm not sure what the problem is since it's all working fine here. I can ListVaults and CreateVault using master.

Can you make sure you have the latest code and try again? Many thanks.

Cheers,
Andy

**Comment by chad-skidmore** [comment]
I only see commits for CreateVault and the CreateVault example. Did you miss pushing a commit?

**Comment by chilts** [comment]
Ah yep, sorry, I meant I did a fix for CreateVault. However, the ListVaults is also fine here (as shown above) and has been since the origin commit of Glacier.

Still no good on master for either ListVaults or CreateVault?

Cheers,
Andy

**Comment by chad-skidmore** [comment]
Ok, I see the issue. The account number assumes with dashes removed rather than as displayed by Amazon. Sorry for not checking that sooner. Silly mistake.

With a corrected account number sans dashes all is good. Will bang away on it and let you know if I run across anything else. I've spent some time walking through your source so I understand your module far better now than I did before.

Thanks again for the quick responses.
